x-pack/plugins/observability_solution/observability_onboarding/server/saved_objects/observability_onboarding_status.ts

```diff
@@ -92,5 +98,24 @@ export const observabilityOnboardingFlow: SavedObjectsType = {
         }),
       },
     },
+    '2': {
+      changes: [],
+      schemas: {
+        create: schema.object({
+          type: schema.string(),
+          state: schema.maybe(schema.oneOf([LogFilesStateSchema, SystemLogsStateSchema])),
```

Comment:
I'm not sure if the schemas in `state` are applicable to the auto detect flow. We are not saving any of those properties and I would generally try and avoid creating a separate schema for each type of integration like the current onboarding flow does. 

The list of integrations we want to support in auto-detect will likely grow and we're only using this saved object to track progress in the UI, rather than capture all selected parameters. 

I think all we need is pretty much the type and progress parts of the current schema:

```
const customLogsSchema = {
   type: 'customLogs',
   state,
   progress
}
const systemLogsSchema = {
   type: 'systemLogs'
   state,
   progress
}
const autoDetectSchema = {
   type: 'autoDetect'
   progress
}

const savedObjectType = {
  schemas: {
    create: schema.oneOf(customLogsSchema, systemLogsSchema, autoDetectSchema)
  }
}
```

Reply:
That's a good point 👍

Reply:
Ah, unfortunately `schema.create` expects an `ObjectType` explicitly, we cannot use `oneOf` there. But I added `schema.never()` to the state schema, so we have an option to skip it for auto detect flow.